design/unnumbered-bgp.md

```diff
@@ -0,0 +1,508 @@
+// 2. if non valid address and no interface then fail
+// 3. if address is LLA and interface not empty, then LLA support
+// == extra FRR directive `neighbor fe80::a876:4dff:fe77:408 interface eth0`
+// 4. If address is LLA and empty interface, then fail
+// 5. If address is empty and interface has value then the BGP unnumber peering config will take place
+// ==  modify FRR directive `neighbor PEER interface remote-as <ASN>`
+// && replace any address directive to use interface e.g.  `neighbor net0 timers 180 540`
+// && enable RAs on this interface
+// 6. If address has value (non-LLA) and interface non empty, then fail
+```
+
+##### FRR-K8S
+
+```go
+type Neighbor struct {
+  Address string `json:"address"`
+  Interface string `json:"interface"`
+}
+```
+
+##### MetalLB
+
+```go
+type BGPPeerSpec struct {
+  Address string `json:"peerAddress"`
+  Interface string `json:"interface"`
+}
+```
+
+Example CR
+
+```yaml
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: peer-unnumber
+  namespace: metallb-system
+spec:
+  myASN: 64512
+  ASN: 645000
+  interface: net0
+```
+
+#### Option 3. Add new Unnumber bool field
+
+We can introduce a new field `Unnumber bool`, which if true the we indicate
+Unnumber BGP Peering. If `true` the Address field holds the interface value.
+Bool option makes the configuration error-free, because the user declares
+explicitly that they want Unnumber and there is no "guessing" what metallb
+does. Unnumber case is disconnected from LLA support case.
+
+##### FRR-K8S
+```go
+
+type Neighbor struct {
+  Address string `json:"address"` // valid IP or interface if unnumber is true
+  Unnumber bool `json:"Unnumber"`
+}
+```
+
+##### MetalLB
+
+```go
+type BGPPeerSpec struct {
+  Address string `json:"peerAddress"`
+  Unnumber bool `json:"unnumber"`
+}
+```
+
+Example CR
+
+```yaml
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: peer-unnumber
+  namespace: metallb-system
+spec:
+  myASN: 64512
+  ASN: 645000
+  peerAddress: net0
+  unnumber: true
+```
+
+### Testing
+
+#### E2E test for [Unnumber on primary interface](design/unnumbered-bgp.md#story-1-unnumbered-iebgp-on-primary-node-interface)
+
+There is no straightforward way to test unnumbered in the primary node
+interface. The testing setup (kind creation) creates bridge and connect
+containers. Ideally we need to setup a kind cluster, and replace the bridge
+with a containers that runs FRR on each other side of the veth, and at the same
+time to provide switch/upstream. Until we find a way to automate that infra setup
+we test that scenario manual.
+
+#### E2E test for [Unnumber on secondary interface](design/unnumbered-bgp.md#story-1-unnumbered-iebgp-on-secondary-node-interface)
+
+We should modify the testing infra to include point-to-point links which means
+connect containers using VETH, and not through a bridge. We need NET-ADMIN cap
+(or sudo) when creating the VETH.
+
+##### Option 1 - have a separate test
```

Comment:
I'd start with this, the majority of the scenarios are functional and should (!!!) be orthogonal to how we establish the session

Reply:
Sounds good, i keep the text as is, one we have option 1 I would say we can merge, but I am in favor to then try to implement also option